jlm/tooling/CommandGraphGenerator.cpp

```diff
@@ -114,99 +114,17 @@ JlcCommandGraphGenerator::GenerateCommandGraph(const JlcCommandLineOptions & com
 
     if (compilation.RequiresOptimization())
     {
-      std::vector<JlmOptCommand::Optimization> optimizations;
-      if (!commandLineOptions.JlmOptOptimizations_.empty())
-      {
-        static std::unordered_map<JlmOptCommandLineOptions::OptimizationId, JlmOptCommand::Optimization> map(
-          {
-            {
-              JlmOptCommandLineOptions::OptimizationId::AASteensgaardAgnostic,
-              JlmOptCommand::Optimization::AASteensgaardAgnostic
-            },
-            {
-              JlmOptCommandLineOptions::OptimizationId::AASteensgaardRegionAware,
-              JlmOptCommand::Optimization::AASteensgaardRegionAware
-            },
-            {
-              JlmOptCommandLineOptions::OptimizationId::cne,
-              JlmOptCommand::Optimization::CommonNodeElimination
-            },
-            {
-              JlmOptCommandLineOptions::OptimizationId::dne,
-              JlmOptCommand::Optimization::DeadNodeElimination
-            },
-            {
-              JlmOptCommandLineOptions::OptimizationId::iln,
-              JlmOptCommand::Optimization::FunctionInlining
-            },
-            {
-              JlmOptCommandLineOptions::OptimizationId::InvariantValueRedirection,
-              JlmOptCommand::Optimization::InvariantValueRedirection
-            },
-            {
-              JlmOptCommandLineOptions::OptimizationId::psh,
-              JlmOptCommand::Optimization::NodePushOut
-            },
-            {
-              JlmOptCommandLineOptions::OptimizationId::pll,
-              JlmOptCommand::Optimization::NodePullIn
-            },
-            {
-              JlmOptCommandLineOptions::OptimizationId::red,
-              JlmOptCommand::Optimization::NodeReduction
-            },
-            {
-              JlmOptCommandLineOptions::OptimizationId::ivt,
-              JlmOptCommand::Optimization::ThetaGammaInversion
-            },
-            {
-              JlmOptCommandLineOptions::OptimizationId::url,
-              JlmOptCommand::Optimization::LoopUnrolling},
-          });
-
-        for (const auto & jlmOpt : commandLineOptions.JlmOptOptimizations_)
-        {
-          JLM_ASSERT(map.find(jlmOpt) != map.end());
-          optimizations.push_back(map[jlmOpt]);
-        }
-
-        /*
-         * If a default optimization level has been specified (-O) and no specific jlm options
-         * have been specified (-J) then use a default set of optimizations.
-         */
-      } else if (commandLineOptions.JlmOptOptimizations_.empty()
-          && commandLineOptions.OptimizationLevel_ == JlcCommandLineOptions::OptimizationLevel::O3)
-      {
-        /*
-         * Only -O3 sets default optimizations
-         */
-        optimizations = {
```

Comment:
I realize that the definition of the OptimizationIds are not nor part of this PR, but the list of optimizations have gone from more descriptive names to the shorthand names, e.g., from DeadNodeElimination to dne

Reply:
Yes, the OptimizationIds of the jlm-opt command line tool have not changed, and yes, they are currently very undescriptive. However, making them more descriptive is for another PR.